Reemit.Gui/Views/Controls/HexEditor/HexEditorView.axaml.cs

```diff
@@ -0,0 +1,108 @@
+{
+    private int? _lastBytesPerLine;
+
+    public HexEditorView()
+    {
+        InitializeComponent();
+
+        this.WhenActivated(d =>
+        {
+            this.OneWayBind(ViewModel, x => x.ModuleDocument, x => x.ReemitHexEditor.Document)
+                .DisposeWith(d);
+
+            this.OneWayBind(ViewModel, x => x.ByteWidths, x => x.HexWidthCombobox.ItemsSource)
+                .DisposeWith(d);
+
+            this.Bind(ViewModel, x => x.CurrentByteWidth, x => x.HexWidthCombobox.SelectedItem)
+                .DisposeWith(d);
+
+            ViewModel
+                .WhenAnyValue(x => x.CurrentByteWidth)
+                .WhereNotNull()
+                .Select(x => x.Width)
+                .BindTo(ReemitHexEditor.HexView, x => x.BytesPerLine)
+                .DisposeWith(d);
+
+            _lastBytesPerLine = ReemitHexEditor.HexView.BytesPerLine;
+
+            ReemitHexEditor.HexView
+                .WhenAnyValue(x => x.BytesPerLine)
+                .Subscribe(_ => BringSelectionIntoView())
+                .DisposeWith(d);
+
+            var selection = ReemitHexEditor.Selection;
+
+            var rangeChanged = Observable
+                .FromEvent<EventHandler, EventArgs>(
+                    h => (o, e) => h(e),
+                    h => selection.RangeChanged += h,
+                    h => selection.RangeChanged -= h);
+
+            rangeChanged
+                .Select(_ => selection.Range)
+                .BindTo(ViewModel, x => x.SelectedRange)
+                .DisposeWith(d);
+
+            rangeChanged
+                .Subscribe(_ => BringSelectionIntoView())
+                .DisposeWith(d);
+
+            ViewModel
+                .WhenAnyValue(x => x.SelectedRange)
+                .WhereNotNull()
+                .BindTo(ReemitHexEditor, x => x.Selection.Range)
+                .DisposeWith(d);
+
+            this.BindCommand(ViewModel, x => x.Navigation.NextCommand, x => x.NavigateNextButton)
+                .DisposeWith(d);
+
+            this.BindCommand(ViewModel, x => x.Navigation.PreviousCommand, x => x.NavigatePreviousButton)
+                .DisposeWith(d);
+
+            this.OneWayBind(ViewModel, x => x.SelectionOffset, x => x.OffsetTextBox.Text);
+            this.OneWayBind(ViewModel, x => x.SelectionEnd, x => x.EndTextBox.Text);
+            this.OneWayBind(ViewModel, x => x.SelectionLength, x => x.LengthTextBox.Text);
+
+            Observable
+                .FromEvent<EventHandler, EventArgs>(
+                    h => (o, e) => h(e),
+                    h => ReemitHexEditor.HexView.LayoutUpdated += h,
+                    h => ReemitHexEditor.HexView.LayoutUpdated -= h)
+                .Where(_ => _lastBytesPerLine != ReemitHexEditor.HexView.BytesPerLine)
+                .Subscribe(_ =>
+                {
+                    _lastBytesPerLine = ReemitHexEditor.HexView.BytesPerLine;
+                    BringSelectionIntoView();
+                });
```

Comment:
Probably should use the `DisposeWith()` extension here.

Reply:
Done. Looking forward, I wonder if this is something that could be identified with some Roslyn analysis.